sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/src/autorest.md

```diff
@@ -70,4 +105,46 @@ directive:
       $.GuestConfigurationNavigation.properties.assignmentType['x-nullable'] = true;
       $.GuestConfigurationNavigation.properties.kind['x-nullable'] = true;
       $.GuestConfigurationAssignmentProperties.properties.vmssVMList['x-nullable'] = true;
+  - from: guestconfiguration.json
+    where: $.definitions
+    transform: >
+      $.Resource = {
+          "description": "The core properties of guest configuration resources",
+          "properties": {
+            "id": {
+              "type": "string",
+              "readOnly": true,
+              "x-ms-format": "arm-id",
+              "description": "ARM resource id of the guest configuration assignment."
+            },
+            "name": {
+              "type": "string",
+              "readOnly": false,
+              "description": "Name of the guest configuration assignment."
+            },
+            "location": {
+              "type": "string",
+              "readOnly": false,
+              "x-ms-fromat": "azure-location",
+              "description": "Region where the VM is located."
+            },
+            "type": {
+              "readOnly": true,
+              "type": "string",
+              "x-ms-format": "resource-type",
+              "description": "The type of the resource."
+            },
+            "systemData": {
+              "readOnly": true,
+              "type": "object",
+              "description": "Azure Resource Manager metadata containing createdBy and modifiedBy information.",
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+            }
+          },
+          "x-ms-azure-resource": true,
+          "x-ms-client-name": "GuestConfigurationResourceData"
+        }
+      $.GuestConfigurationAssignment.allOf[0]['$ref'] = '#/definitions/Resource';
```

Comment:
why not just ref the `Resource` in the `common-types`?

Reply:
Guest configuration defines its own resource model and it has a settable name and an additional property `location`, it's different from our common type, therefore I didn't reference the common type here.